### PR TITLE
Add PR build workflow.

### PR DESCRIPTION
**Pull request by anuraaga**
- Replaces github api plugin with a github contributors one. For some reason, former doesn't run with `secrets.GITHUB_TOKEN` which is available in PR builds
- Removes filtering of contributors to only org members which seems like what we want
- Checks in package-lock.json as best practice
- Uploads site built during PR build which can be used for verification of the output before merge if interesting

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       - uses: actions/checkout@v1
       - uses: enriikke/gatsby-gh-pages-action@v2
         env:
-          GH_API_KEY: ${{ secrets.gh }}
+          GH_API_KEY: ${{ secrets.GITHUB_TOKEN }}
           GA_KEY: ${{ secrets.ga }}
         with:
           access-token: ${{ secrets.demo }}
```
Comment (anuraaga) [inline .github/workflows/main.yml:18]: I think GITHUB_TOKEN is enough for this too but will send a separate PR

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
```
Comment (mxiamxia) [inline .github/workflows/pr.yml:22]: add `check-latest: true`?

File: package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "gatsby-starter-rocket-docs",
+  "name": "aws-otel-docs",
```
Comment (mxiamxia) [inline package.json:2]: 👍

**Comment by mxiamxia** [comment]
I'll approve when the conflicts are resolved. Thanks.

**Comment by anuraaga** [comment]
Thanks!

**Review by mxiamxia — APPROVED** [review]
